modules/renter/contractor/editor.go

```diff
@@ -117,12 +117,18 @@ func (he *hostEditor) Upload(data []byte) (crypto.Hash, error) {
 	uploadDelta := he.editor.UploadSpending.Sub(oldUploadSpending)
 	storageDelta := he.editor.StorageSpending.Sub(oldStorageSpending)
 
-	he.contractor.mu.Lock()
-	he.contractor.financialMetrics.UploadSpending = he.contractor.financialMetrics.UploadSpending.Add(uploadDelta)
-	he.contractor.financialMetrics.StorageSpending = he.contractor.financialMetrics.StorageSpending.Add(storageDelta)
-	he.contractor.contracts[contract.ID] = contract
-	he.contractor.saveSync()
-	he.contractor.mu.Unlock()
+	c := he.contractor
+	c.mu.Lock()
+	metrics := c.contractMetrics[contract.ID]
+	metrics.UploadSpending = metrics.UploadSpending.Add(uploadDelta)
+	metrics.StorageSpending = metrics.StorageSpending.Add(storageDelta)
+	metrics.Unspent = metrics.Unspent.Sub(uploadDelta.Add(storageDelta))
```

Comment:
I think it would be less confusing to do `metrics.Unspend.Sub(ud).Sub(sd)` -> seeing `Add(storageDelta)` is confusing when you are expecting the value to be subtracted out.